engine-nextgen/src/main/kotlin/graphql/nadel/enginekt/blueprint/NadelExecutionBlueprintFactory.kt

```diff
@@ -280,8 +296,9 @@ private class Factory(
                     val pathToField = remoteArgDef.remoteArgumentSource.path
                     NadelHydrationActorInputDef.ValueSource.FieldResultValue(
                         queryPathToField = NadelQueryPath(pathToField),
-                        fieldDefinition = getUnderlyingType(hydratedFieldParentType)
-                            ?.getFieldAt(pathToField)
+                        fieldDefinition = mapFirstNotNullFromUnderlyingType(hydratedFieldParentType) { underlyingParentType ->
```

Comment:
We're getting the first underlying type that has `getFieldAt(pathToField)` as NOT null and fetching that value.